libvast/src/format/arrow.cpp

```diff
@@ -74,4 +75,109 @@ bool writer::layout(const std::shared_ptr<::arrow::Schema>& schema) {
   return false;
 }
 
+arrow_istream_wrapper::arrow_istream_wrapper(std::shared_ptr<std::istream> input)
+  : input_(std::move(input)), pos_(0) {
+  set_mode(::arrow::io::FileMode::READ);
+}
+
+::arrow::Status arrow_istream_wrapper::Close() {
+  // TODO: is there a way to actually close an istream?
```

Comment:
Since `input_` is just a shared pointer here you may as well check whether it's a nullptr and use that for signaling whether the stream is closed.

Reply:
I changed the method to set `input_ = nullptr` on `Close()` and check for `input_` on `closed()`. Hope that does the trick.